### PR TITLE
Migrate automatically

### PR DESCRIPTION
**Pull request by Szymon20000**
## Description

Automatically use interpolateNode nad EasingNode when users still use old naming (interpolate, Easing) in V1 code.

## Test code and steps to reproduce

```JS
import React from "react";
import { SectionList, StatusBar, StyleSheet, Text, TouchableOpacity, View } from "react-native";
import Animated, { Easing } from 'react-native-reanimated';

const App = () => {
  let opacity = new Animated.Value(0);

  const animate = easing => {
    opacity.setValue(0);
    Animated.timing(opacity, {
      toValue: 1,
      duration: 1200,
      easing
    }).start();
  };

  const size = opacity.interpolate({
    inputRange: [0, 1],
    outputRange: [0, 80]
  });

  const animatedStyles = [
    styles.box,
    {
      opacity,
      width: size,
      height: size
    }
  ];

  return (
    <View style={styles.container}>
      <StatusBar hidden={true} />
      <Text style={styles.title}>
        Press rows below to preview the Easing!
      </Text>
      <View style={styles.boxContainer}>
        <Animated.View style={animatedStyles} />
      </View>
      <SectionList
        style={styles.list}
        sections={SECTIONS}
        keyExtractor={(item) => item.title}
        renderItem={({ item }) => (
          <TouchableOpacity
            onPress={() => animate(item.easing)}
            style={styles.listRow}
          >
            <Text>{item.title}</Text>
          </TouchableOpacity>
        )}
        renderSectionHeader={({ section: { title } }) => (
          <Text style={styles.listHeader}>{title}</Text>
        )}
      />
    </View>
  );
};

const SECTIONS = [
  {
    title: "Predefined animations",
    data: [
      { title: "Bounce", easing: Easing.bounce },
      { title: "Ease", easing: Easing.ease },
      { title: "Elastic", easing: Easing.elastic(4) }
    ]
  },
  {
    title: "Standard functions",
    data: [
      { title: "Linear", easing: Easing.linear },
      { title: "Quad", easing: Easing.quad },
      { title: "Cubic", easing: Easing.cubic }
    ]
  },
  {
    title: "Additional functions",
    data: [
      {
        title: "Bezier",
        easing: Easing.bezier(0, 2, 1, -1)
      },
      { title: "Circle", easing: Easing.circle },
      { title: "Sin", easing: Easing.sin },
      { title: "Exp", easing: Easing.exp }
    ]
  },
  {
    title: "Combinations",
    data: [
      {
        title: "In + Bounce",
        easing: Easing.in(Easing.bounce)
      },
      {
        title: "Out + Exp",
        easing: Easing.out(Easing.exp)
      },
      {
        title: "InOut + Elastic",
        easing: Easing.inOut(Easing.elastic(1))
      }
    ]
  }
];

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: "#20232a"
  },
  title: {
    marginTop: 10,
    textAlign: "center",
    color: "#61dafb"
  },
  boxContainer: {
    height: 160,
    alignItems: "center"
  },
  box: {
    marginTop: 32,
    borderRadius: 4,
    backgroundColor: "#61dafb"
  },
  list: {
    backgroundColor: "#fff"
  },
  listHeader: {
    paddingHorizontal: 8,
    paddingVertical: 4,
    backgroundColor: "#f4f4f4",
    color: "#999",
    fontSize: 12,
    textTransform: "uppercase"
  },
  listRow: {
    padding: 8
  }
});

export default App;
```

## Checklist

- [ ] Included code example that can be used to test this change
- [ ] Updated TS types
- [ ] Added TS types tests
- [ ] Added unit / integration tests
- [ ] Updated documentation
- [ ] Ensured that CI passes

### CODE REVIEW

File: src/reanimated2/interpolation.js [hunk 1]
```diff
@@ -112,9 +114,7 @@ function internalInterpolate(x, l, r, ll, rr, type) {
 export function interpolate(x, input, output, type) {
   'worklet';
   if (x && x.__nodeID) {
-    throw new Error(
```
Comment (jakub-gonet) [inline src/reanimated2/interpolation.js:115]: I'd leave here console.warn.

Comment (Szymon20000) [reply src/reanimated2/interpolation.js:115]: Like those ones?

Comment (jakub-gonet) [reply src/reanimated2/interpolation.js:115]: I'd go with something like:
"interpolate() was renamed to interpolateNode() in Reanimated 2. Please use interpolateNode() instead".

**Review by kmagiera — APPROVED** [review]

File: src/reanimated2/Easing.js [hunk 2]
```diff
@@ -252,3 +254,39 @@ export const Easing = {
   out,
   inOut,
 };
+
+function createChecker(worklet, workletName, prevArgs) {
+  function checkIfReaOne() {
+    'worklet'
+    if (arguments && (!_WORKLET)) {
+      for (let i = 0; i < arguments.length; i++) {
+        const arg = arguments[i];
```
[the remark this answers is not in the document]
Comment (jakub-gonet) [reply src/reanimated2/Easing.js:263]: Nvm, this is worklet.

File: src/reanimated2/Easing.js [hunk 3]
```diff
@@ -252,3 +254,39 @@ export const Easing = {
...
+  checkIfReaOne.__location = worklet.__location;
+  return checkIfReaOne;
+}
+
+Object.keys(EasingObject).forEach((key) => {
+  EasingObject[key] = createChecker(EasingObject[key], key);
+});
+
+export const Easing = EasingObject;
+export default Easing;
```
[the remark this answers is not in the document]
Comment (Szymon20000) [reply src/reanimated2/Easing.js:292]: We don't.

**Pull request merged**